### DISCUSSION TITLE
Agent for multiple Projects

### DISCUSSION THREAD

**Discussion by sti0** [PrefectHQ/prefect#4052] [Q&A] [answered]
Hi there,

I'm using latest 0.14.6 release with an on-premise installation. 

I set up two projects and one agent. If I trigger my flows the agent only runs the flows from project 1. The flows in project 2 are still scheduled and become late.

Is it necessary to set up an second agent? How to allocate the agent to the project? Or is it a bug?

Thanks in advance :)

**Comment by zanieb** [comment, marked as answer]
Hi @sti0 

Agents shouldn't have any concept of projects, they should just be filtering your flows based on labels. You'll have to provide some more information about your setup. I'd recommend reading through https://docs.prefect.io/orchestration/agents/overview.html

**Reply by sti0** [reply]
Started from scratch and it works now. Don't know what happend. This is not reproducible.

**Reply by moisesrc13** [reply]
@madkinsz  if I have a flow with multiple labels matching multiple local agents, how is the agent selected when I start the flow?, which criteria is used to select the agent?, which is the underlying prefect process taking care of this?

**Reply by temirovazat** [reply]
Я бы рекомендовал прочитать [https://docs.prefect.io/orchestration/agents/overview.html.](https://docs.prefect.io/orchestration/agents/overview.html)

**Reply by moisesrc13** [reply]
@temirovazat thanks for the link, however, still don't see in the documentation what happens in the scenario where I have registered a flow using multiple labels from multiple local agents, so when I start the flow, how is the agent selected by the prefect api?. I did this for scaling and redundancy, I don't really need to run different flows on different agents due to runtime constraints, what I need is to be able to run a flow on any agent at any given time, if for example, one agent is running already multiple process and there is no resource capacity (cpu, mem)

**Comment by temirovazat** [comment]
Я бы рекомендовал прочитать [https://docs.prefect.io/orchestration/agents/overview.html.](https://docs.prefect.io/orchestration/agents/overview.html)
